core/unit_test/tstDeepCopy.hpp

```diff
@@ -176,41 +176,83 @@ void testMirror()
                     slice_3( idx, i, j ) = dval * (i+j);
         });
 
-    // Create a mirror with the same memory space and copy.
-    auto same_space_copy = Cabana::Experimental::create_mirror_view_and_copy(
-        TEST_MEMSPACE(), aosoa );
-    EXPECT_EQ( same_space_copy.size(), aosoa.size() );
-    bool ssc_same_ms =
-        std::is_same<TEST_MEMSPACE,
-                     decltype(same_space_copy)::memory_space>::value;
-    EXPECT_TRUE( ssc_same_ms );
-    bool ssc_same_mt =
-        std::is_same<DataTypes,
-                     decltype(same_space_copy)::member_types>::value;
-    EXPECT_TRUE( ssc_same_mt );
-
-    // Check that the same memory space case didn't allocate any memory. They
-    // should have the same pointer.
-    EXPECT_EQ( aosoa.ptr(), same_space_copy.ptr() );
-
-    // Check values.
-    checkDataMembers( same_space_copy, fval, dval, ival, dim_1, dim_2, dim_3 );
-
-    // Create a mirror with the host space and copy.
-    auto host_space_copy = Cabana::Experimental::create_mirror_view_and_copy(
-        Kokkos::HostSpace(), aosoa );
-    EXPECT_EQ( host_space_copy.size(), aosoa.size() );
-    bool hsc_same_ms =
-        std::is_same<Kokkos::HostSpace,
-                     decltype(host_space_copy)::memory_space>::value;
-    EXPECT_TRUE( hsc_same_ms );
-    bool hsc_same_mt =
-        std::is_same<DataTypes,
-                     decltype(host_space_copy)::member_types>::value;
-    EXPECT_TRUE( hsc_same_mt );
+    // Create a mirror with the same memory space and copy separately.
+    {
+        auto same_space_copy = Cabana::Experimental::create_mirror_view(
+            TEST_MEMSPACE(), aosoa );
+        Cabana::deep_copy( same_space_copy, aosoa );
```

Comment:
Avoid duplicating the blocks.  Encapsulate the "create mirror view and copy" and call the test for the two variants.

Reply:
fixed

Reply:
well - I did a variant of what you suggested that reused code